src/pkg/cli/compose.go

```diff
@@ -209,26 +263,16 @@ func convertPort(port types.ServicePortConfig) (*v1.Port, error) {
 		return nil, fmt.Errorf("port protocol not one of [tcp udp http http2 grpc]: %v", port.Protocol)
 	}
 
-	logrus := logrus.WithField("target", port.Target)
-
 	switch port.Mode {
-	case "":
-		logrus.Warn("No port mode was specified; assuming 'host' (add 'mode' to silence)")
-		HadWarnings = true
-		fallthrough
-	case "host":
+	case "", "host":
```

Comment:
I understand that `validatePorts` does all the error/warning and `convertPorts` the actual conversion, but now the logic got split up. For example, `case ""` being "host" was explained by the warning, but now it has lost the context. I wonder if we should leave the warnings in "convertPorts", ie. only move the errors to "validatePorts"

Reply:
Good point, I think letting convert to do the warning is not a bad idea, clearer separation of responsibility, updated.